apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/interceptor/enhance/DelegateNamingResolver.java

```diff
@@ -0,0 +1,79 @@
+import java.util.Objects;
+
+/**
+ * Generate fixed delegate field name for MethodDelegation
+ */
+public class DelegateNamingResolver {
+    private static final String PREFIX = "delegate$";
+    private final String className;
+    private final int identifier;
+    private final String fieldNamePrefix;
+
+    public DelegateNamingResolver(String className, int identifier) {
+        this.className = className;
+        this.identifier = identifier;
+        // Interceptor delegate field name pattern: <name_trait>$delegate$<class_name_hash>$<plugin_define_hash>$<intercept_point_hash>
+        // something like: InstMethodsInter sw$delegate$td03673$sib0lj0$5n874b1;
+        this.fieldNamePrefix = Constants.NAME_TRAIT + PREFIX + RandomString.hashOf(className.hashCode()) + "$" + RandomString.hashOf(identifier) + "$";
+    }
+
+    public String resolve(Object interceptPoint) {
+        Objects.requireNonNull(interceptPoint, "interceptPoint cannot be null");
+        return fieldNamePrefix + RandomString.hashOf(computeHashCode(interceptPoint));
+    }
+
+    private int computeHashCode(Object interceptPoint) {
+        String interceptPointClassName = interceptPoint.getClass().getName();
+        if (interceptPoint instanceof ConstructorInterceptPoint) {
+            ConstructorInterceptPoint point = (ConstructorInterceptPoint) interceptPoint;
+            return Objects.hash(interceptPointClassName, point.getConstructorMatcher().toString(), point.getConstructorInterceptor());
+        } else if (interceptPoint instanceof InstanceMethodsInterceptPoint) {
+            InstanceMethodsInterceptPoint point = (InstanceMethodsInterceptPoint) interceptPoint;
+            return Objects.hash(interceptPointClassName, point.getMethodsMatcher().toString(), point.getMethodsInterceptor(), point.isOverrideArgs());
+        } else if (interceptPoint instanceof InstanceMethodsInterceptV2Point) {
+            InstanceMethodsInterceptV2Point point = (InstanceMethodsInterceptV2Point) interceptPoint;
+            return Objects.hash(interceptPointClassName, point.getMethodsMatcher().toString(), point.getMethodsInterceptorV2(), point.isOverrideArgs());
+        } else if (interceptPoint instanceof StaticMethodsInterceptPoint) {
+            StaticMethodsInterceptPoint point = (StaticMethodsInterceptPoint) interceptPoint;
+            return Objects.hash(interceptPointClassName, point.getMethodsMatcher().toString(), point.getMethodsInterceptor(), point.isOverrideArgs());
+        } else if (interceptPoint instanceof StaticMethodsInterceptV2Point) {
+            StaticMethodsInterceptV2Point point = (StaticMethodsInterceptV2Point) interceptPoint;
+            return Objects.hash(interceptPointClassName, point.getMethodsMatcher().toString(), point.getMethodsInterceptorV2(), point.isOverrideArgs());
+        }
+        return interceptPoint.hashCode();
+    }
+
+    public static DelegateNamingResolver get(String className, AbstractClassEnhancePluginDefine pluginDefine) {
```

Comment:
How about using a constructor directly? There is no cache mechanism here, and you always create a new one.